src/runtime.js

```diff
@@ -525,13 +523,12 @@ function prepareAndRegisterServiceForDoc(global, extensions,
  * @param {!Window} global
  * @param {!./service/extensions-impl.Extensions} extensions
  * @param {string} name
- * @param {function(new:Object, !./service/ampdoc-impl.AmpDoc)=} opt_ctor
- * @param {function(!./service/ampdoc-impl.AmpDoc):!Object=} opt_factory
+ * @param {!function(new:Object, !./service/ampdoc-impl.AmpDoc)} ctor
```

Comment:
Can we figure out how to make this a bi-directional code in `runtime.js` only? Ideally, `service` doesn't need to do that. Or at the worst, you may need to preserve `build()` function that you had before?

Ping me to discuss if needed. It is somewhat important. It should also come with `TODO` and a cleanup task.